DifferentLocation.java

```diff
@@ -0,0 +1,120 @@
+package location;
+
+import java.util.List;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.support.ui.Select;
+
+public class DifferentLocation {
+
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+		WebDriver driver = new FirefoxDriver();
+
+		String appUrl = "http://www.tripfactory.com/";
+		driver.get(appUrl);
+
+		// Click on Location
+
+		driver.findElement(By.linkText("Location")).click();
+
+		List<WebElement> allElements = driver.findElements(By.xpath("//div[1]/div[3]/div/nav/ul/li[5]/ul"));
+
+		for (WebElement element : allElements) {
+			System.out.println(element.getText());
+		}

```

Comment:
Why do you have to use the String "Australia" at all in your code ?
Figure out a way to access the element from the list, without needing to
look it up again instead of using findElements again for Australia. You
already have that element in `allElements`


Nitish Varma

CTO and Head - Payments

www.tripfactory.com

<https://twitter.com/_tripfactory> ​​
<https://www.facebook.com/tripfactorycom>


On Fri, Mar 18, 2016 at 3:16 PM, aakashsehrawat <notifications@github.com>
wrote:

> In DifferentLocation.java
> <https://github.com/aakashsehrawat/automation/pull/1#discussion_r56632727>
> :
>
> > +	public static void main(String[] args) {
> > +		// TODO Auto-generated method stub
> > +		WebDriver driver = new FirefoxDriver();
> > +
> > +		String appUrl = "http://www.tripfactory.com/";
> > +		driver.get(appUrl);
> > +
> > +		// Click on Location
> > +
> > +		driver.findElement(By.linkText("Location")).click();
> > +
> > +		List<WebElement> allElements = driver.findElements(By.xpath("//div[1]/div[3]/div/nav/ul/li[5]/ul"));
> > +
> > +		for (WebElement element : allElements) {
> > +			System.out.println(element.getText());
> > +		}
>
> And what if I add an If statement here like:
> if(element.getText().equals("Australia")).click; ?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/aakashsehrawat/automation/pull/1/files/e57ea05df23abf42cae4eb5cec6fa2389d450dc1#r56632727>
>